src/Grassmann.jl

```diff
@@ -249,8 +253,33 @@ function __init__()
             generate_derivation(:(Reduce.Algebra),T,:df,:RExpr)
         end
     end
-    @require SymPy="24249f21-da20-56a4-8eb1-6a02cf4ae2e6" generate_algebra(:SymPy,:Sym,:diff,:symbols)
-    @require SymEngine="123dc426-2d89-5057-bbad-38513e3affd8" generate_algebra(:SymEngine,:Basic,:diff,:symbols)
+    @require SymPy="24249f21-da20-56a4-8eb1-6a02cf4ae2e6" begin
+        generate_algebra(:SymPy,:Sym,:diff,:symbols)
+        T = Union{TensorGraded, TensorMixed}
+        symbolic = v -> typeof(v) == SymPy.Sym
+        SymPy.expand(x::T) = map(v -> symbolic(v) ? SymPy.expand(v) : v, x)
+        SymPy.factor(x::T) = map(v -> symbolic(v) ? SymPy.factor(v) : v, x)
+        SymPy.together(x::T) = map(v -> symbolic(v) ? SymPy.together(v) : v, x)
+        SymPy.apart(x::T) = map(v -> symbolic(v) ? SymPy.apart(v) : v, x)
+        SymPy.cancel(x::T) = map(v -> symbolic(v) ? SymPy.cancel(v) : v, x)
+        # SymPy.expand_trig(x::T) = map(v -> symbolic(v) ? SymPy.expand_trig(v) : v, x)
+        SymPy.N(x::T, args...) = map(v -> symbolic(v) ? SymPy.N(v, args...) : v, x)
+        # SymPy.evalf(X::T, args...) = map(v -> symbolic(v) ? SymPy.evalf(v, args...) : v, x)
+        SymPy.convert(X::T, args...) = map(v -> symbolic(v) ? SymPy.convert(v, args...) : v, x)
+        SymPy.subs(x::T, args...) = map(v -> symbolic(v) ? SymPy.subs(v, args...) : v, x)
+        SymPy.collect(x::T, args...) = map(v -> symbolic(v) ? SymPy.collect(v, args...) : v, x)
```

Comment:
This is type piracy, hijacking global `convert` method and `collect` method for `T`.

You'll have to come up with another solution here as well.

Reply:
Hmm!  My goal was to provide an interface to the SymPy `convert` and `collect` methods [here](https://github.com/JuliaPy/SymPy.jl/blob/master/examples/tutorial.md#the-expand-factor-collect-and-simplify-functions) and [here](https://github.com/JuliaPy/SymPy.jl/blob/master/src/numbers.jl#L11-L34)

Would what `Sympy.jl` and `SymEnhine.jl` be doing here also be considered type piracy then?

Reply:
Instead, what you need is `Chain{V,G,Sym} where {V,G}` or `MultiVector{V,Sym} where V`.

You are over-riding the global `convert` methods. You need to restrict yourself to a subset of elements that is actually relevant to what you are trying to accomplish.